mmedit/datasets/pipelines/generate_assistant.py

```diff
@@ -5,6 +5,78 @@
 from .utils import make_coord
 
 
+@PIPELINES.register_module()
+class GenerateHeatmap:
+    """Generate heatmap from keypoint.
+
+    Args:
+        keypoint (str): Key of keypoint in dict.
+        ori_size (int | Tuple[int]): Original image size of keypoint.
+        target_size (int | Tuple[int]): Target size of heatmap.
+        sigma (float): Sigma parameter of heatmap. Default: 1.0
+    """
+
+    def __init__(self, keypoint, ori_size, target_size, sigma=1.0):
+        if isinstance(ori_size, int):
+            ori_size = (ori_size, ori_size)
+        else:
+            ori_size = ori_size[:2]
+        if isinstance(target_size, int):
+            target_size = (target_size, target_size)
+        else:
+            target_size = target_size[:2]
+        self.rate = (target_size[0] / ori_size[0],
+                     target_size[1] / ori_size[1])
+        self.keypoint = keypoint
+        self.sigma = sigma
+        self.target_size = target_size
+
+    def __call__(self, results):
+        """Call function.
+
+        Args:
+            results (dict): A dict containing the necessary information and
+                data for augmentation. Require keypoint.
+
+        Returns:
+            dict: A dict containing the processed data and information.
+                Add 'heatmap'.
+        """
+        keypoint_list = [(keypoint[0] * self.rate[0],
+                          keypoint[1] * self.rate[1])
+                         for keypoint in results[self.keypoint]]
+        heatmap_list = [
+            self._generate_one_heatmap(keypoint) for keypoint in keypoint_list
+        ]
+        results['heatmap'] = np.stack(heatmap_list, axis=2)
+        return results
+
+    def _generate_one_heatmap(self, keypoint):
+        """Generate One Heatmap.
+
+        Args:
+            landmark (Tuple[float]): Location of a landmark.
+
+        results:
+            heatmap (np.ndarray): A heatmap of landmark.
+        """
+        w, h = self.target_size
+
+        x_range = np.arange(start=0, stop=w, dtype=int)
+        y_range = np.arange(start=0, stop=h, dtype=int)
+        xx, yy = np.meshgrid(x_range, y_range)
+        d2 = (xx - keypoint[0])**2 + (yy - keypoint[1])**2
```

Comment:
`d2` -> `dist`?

Reply:
Actually here d2 is better than dist, since it is distance squared.